src/net/gateway-archive.ts

```diff
@@ -0,0 +1,90 @@
+import { FileApi } from "../api/file";
+import { ContentUri } from "../wrappers/content-uri"
+import {
+    VOCDONI_MAINNET_ENTITY_ID,VOCDONI_RINKEBY_ENTITY_ID, VOCDONI_GOERLI_ENTITY_ID, VOCDONI_XDAI_ENTITY_ID, VOCDONI_SOKOL_ENTITY_ID, XDAI_ENS_REGISTRY_ADDRESS, XDAI_PROVIDER_URI, XDAI_CHAIN_ID,
+    SOKOL_CHAIN_ID, SOKOL_PROVIDER_URI, SOKOL_ENS_REGISTRY_ADDRESS, XDAI_STG_ENS_REGISTRY_ADDRESS, VOCDONI_XDAI_STG_ENTITY_ID
+} from "../constants"
+import { TextRecordKeys } from "../models/entity"
+import { IGateway } from "./gateway";
+import { DVoteGatewayResponseBody } from "./gateway-dvote";
+import { IGatewayPool } from "./gateway-pool";
+import { keccak256 } from "@ethersproject/keccak256"
+
+export type EthNetworkID = "mainnet" | "rinkeby" | "goerli" | "xdai" | "sokol"
+
+export class GatewayArchive {
+
+    /**
+     * Fetch the process data on the IPFS archive
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    public static getProcessFromArchive(processId: string, gateway: IGateway | IGatewayPool, errorMessage: string): Promise<DVoteGatewayResponseBody> {
+        return this.getArchiveUri(gateway)
+            .then((archiveUri: ContentUri) => {
+                return this.getArchiveFile(archiveUri, processId, gateway)
+            })
+            .then((result: string) => {
+                return JSON.parse(result)
+            })
+            .catch((error) => {
```

Comment:
```suggestion
            .catch(() => new Error(errorMessage))
```

Assuming that we don't create a custom Error class for this